### DISCUSSION TITLE
Link tracking using JavaScript

### DISCUSSION THREAD

**Discussion by nandedamana** [EFForg/privacybadger#2757] [General]
Privacy Badger being able to unwrap Facebook and Google redirect links is great. But what about link tracking implemented using JavaScript? I mean, what if a site tracks link activity using click event listeners and beacons? Has it been observed? Has it been handled already?

**Comment by ghostwords** [comment]
Hello!

Sites tend to use third-party analytics services such as Google Analytics. Privacy Badger should already know to block all common analytics services, as part of Privacy Badger's mission of stopping non-consensual [third-party tracking](https://privacybadger.org/#What-is-a-third-party-tracker).

We added some first-party anti-tracking measures specifically for [Facebook](https://www.eff.org/deeplinks/2018/05/privacy-badger-rolls-out-new-ways-fight-facebook-tracking) and [Google](https://www.eff.org/deeplinks/2018/10/privacy-badger-now-fights-more-sneaky-google-tracking) because of the size of those platforms. These are exceptional cases, however, as Privacy Badger's main focus is on stopping pervasive third-party tracking.

Privacy Badger doesn't yet specifically look for beacons when learning what should be blocked. We may add beacon detection in a future update (#2024). But again, common third-party analytics services should already be getting blocked because of Privacy Badger's [existing tracking detection heuristics](https://privacybadger.org/#How-does-Privacy-Badger-work).

Let me know if you have any questions.

**Comment by nandedamana** [comment]
Thanks! This answers my question.
